File: backend/goat_catalog/src/goat_catalog/tokens/otps/application/dto/generate_otp_request.py
"""DTO para solicitud de generación de OTP."""

from pydantic import BaseModel, Field

from ...domain.value_objects.otp_purpose import OTPPurpose


class GenerateOTPRequest(BaseModel):
    """Request para generar un nuevo OTP."""

    email: str = Field(..., description="Email del usuario", examples=["user@example.com"])
    purpose: OTPPurpose = Field(
        ..., description="Propósito del OTP", examples=[OTPPurpose.REGISTER]
    )

    model_config = {"json_schema_extra": {"examples": [{"email": "user@example.com", "purpose": "REGISTER"}]}}

File: backend/goat_listing/src/main/java/com/goat/identity/application/dto/GenerateOtpRequest.java
package com.goat.identity.application.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

/**
 * DTO para la solicitud de generación de OTP.
 */
public class GenerateOtpRequest {
    @NotBlank(message = "El email es requerido")
    @Email(message = "El formato del email no es válido")
    private String email;

    @NotBlank(message = "El propósito es requerido")
    @Pattern(regexp = "REGISTER|LOGIN|EMAIL_CONFIRMATION|RESET_PASSWORD", 
             message = "El propósito debe ser: REGISTER, LOGIN, EMAIL_CONFIRMATION o RESET_PASSWORD")
    private String purpose;

    public GenerateOtpRequest() {
    }

    public GenerateOtpRequest(String email, String purpose) {
        this.email = email;
        this.purpose = purpose;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPurpose() {
        return purpose;
    }

    public void setPurpose(String purpose) {
        this.purpose = purpose;
    }
}

